src/bin/bobd.rs

```diff
@@ -32,26 +38,52 @@ impl server::BobApi for BobSrv {
         type PingFuture = future::FutureResult<Response<Null>, tower_grpc::Status>;
 
         fn put(&mut self, req: Request<PutRequest>) -> Self::PutFuture {
-            let mut param = req.into_inner();
-            Box::new(self.grinder.put(
-                BobKey{
-                    key: param.key.as_ref().unwrap().key
-                    },
-                BobData{
-                    data: param.data.take().unwrap().data
-                },
-                {
-                    let mut opts: BobOptions = Default::default();
-                    if let Some(vopts) = param.options.as_ref() { 
-                        if vopts.force_node {
-                            opts |= BobOptions::FORCE_NODE;
+            let sw = Stopwatch::start_new();
+            let param = req.into_inner();
+
+            if !self.put_validate(&param) {
+                warn!("PUT[-] invalid arguments - key and data is mandatory");
+                Box::new(future::err(tower_grpc::Status::new(tower_grpc::Code::InvalidArgument, "Key and data is mandatory")))
```

Comment:
I would prefer to see the `return` expression here. It is easy to make a mistake with this functional style by  simply adding a code after the `if` statement.

Reply:
This is rust style) and clippy (rust linter) will generate warning of needless return